test/Octonion.jl

```diff
@@ -0,0 +1,146 @@
+        @test convert(Octonion{Float64}, Complex(1, 2)) ===
+            Octonion(1.0, 2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0)
+        @test convert(Octonion{Float64}, Quaternion(1, 2, 3, 4)) ===
+            Octonion(1.0, 2.0, 3.0, 4.0, 0.0, 0.0, 0.0, 0.0)
+        @test convert(Octonion{Float64}, Octonion(1, 2, 3, 4, 5, 6, 7, 8)) ===
+            Octonion(1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0)
+    end
+
+    @testset "promote" begin
+        @test Quaternion(1, 2, 3, 4) == Octonion(1, 2, 3, 4, 0, 0, 0, 0)
+        @test Quaternion(1, 2, 3, 4) != Octonion(1, 2, 3, 4, 5, 6, 7, 8)
+        @test Octonion(1) == 1.0
+        @test Octonion(Complex(1, 2)) == Complex(1, 2)
+    end
+
+    @testset "shorthands" begin
+        @test octo(1) === Octonion(1) # checking the .norm field in particular
+        @test octo(1, 0, 0, 0, 0, 0, 0, 0) === Octonion(1, 0, 0, 0, 0, 0, 0, 0) # checking the .norm field in particular
+        @test octo(1, 2, 3, 4, 5, 6, 7, 8) === Octonion(1, 2, 3, 4, 5, 6, 7, 8)
+        @test octo(Octonion(1, 0, 0, 0, 0, 0, 0, 0)) === Octonion(1, 0, 0, 0, 0, 0, 0, 0) # checking the .norm field in particular
+        @test octo(Octonion(1, 2, 3, 4, 5, 6, 7, 8)) === Octonion(1, 2, 3, 4, 5, 6, 7, 8)
+        @test octo(1, 0, 0, 0, 0, 0, 0, 0, false).norm == false # respect the .norm input (even if wrong)
+        @test octo(1, 2, 3, 4, 5, 6, 7, 8, true).norm == true # respect the .norm input (even if wrong)
+    end
+
+    @testset "random generation" begin
+        @testset "rand($H)" for H in (OctonionF32, OctonionF64)
+            rng = Random.MersenneTwister(42)
+            o = rand(rng, H)
+            @test o isa H
+            @test !o.norm
+
+            os = rand(rng, H, 1000)
+            @test eltype(os) === H
+            @test length(os) == 1000
+            xs = map(os) do o
+                return [real(o); Quaternions.imag(o)]
+            end
+            xs_mean = sum(xs) / length(xs)
+            xs_var = sum(x -> abs2.(x .- xs_mean), xs) / (length(xs) - 1)
+            @test all(isapprox.(xs_mean, 0.5; atol=0.1))
+            @test all(isapprox.(xs_var, 1 / 12; atol=0.01))
+        end
+
+        @testset "randn($H)" for H in (OctonionF32, OctonionF64)
+            rng = Random.MersenneTwister(42)
+            o = randn(rng, H)
+            @test o isa H
+            @test !o.norm
+
+            os = randn(rng, H, 10000)
+            @test eltype(os) === H
+            @test length(os) == 10000
+            xs = map(os) do o
+                return [real(o); Quaternions.imag(o)]
+            end
+            xs_mean = sum(xs) / length(xs)
+            xs_var = sum(x -> abs2.(x .- xs_mean), xs) / (length(xs) - 1)
+            @test all(isapprox.(xs_mean, 0; atol=0.1))
+            @test all(isapprox.(xs_var, 1 / 8; atol=0.1))
+        end
+    end
+
+    @testset "basic" begin
+        real
+        imag
+        conj
+        float
+        Quaternions.abs_imag
+    end
```

Comment:
Is this forgotten to add `@test`?

Reply:
I left these here to note which functions should be tested here, but I've replaced this with a #TODO comment